console/lib/address_ctrl.js

```diff
@@ -15,46 +15,50 @@
  */
 'use strict';
 
-var http = require('http');
+var fs = require("fs");
 var Promise = require('bluebird');
 
-function hostport(defaults) {
-    var result = defaults;
-    if (process.env.ADDRESS_SPACE_SERVICE_HOST) {
-        result.host = process.env.ADDRESS_SPACE_SERVICE_HOST;
-    } else if (process.env.ADDRESS_CONTROLLER_SERVICE_HOST) {
-        result.host = process.env.ADDRESS_CONTROLLER_SERVICE_HOST;
-        if (process.env.ADDRESS_CONTROLLER_SERVICE_PORT_HTTP) {
-            result.port = process.env.ADDRESS_CONTROLLER_SERVICE_PORT_HTTP;
-        }
+var AddressCtrl = function (host, port, ca_file) {
+    this.host = host;
+    this.port = port;
+    this.ca_file = ca_file;
+    this.address_space = process.env.ADDRESS_SPACE
+    this.http = ca_file !== undefined ? require('https') : require('http');
+    this.auth_string = undefined;
+    if (process.env.ADDRESS_SPACE_PASSWORD_FILE) {
+        this.auth_string = 'Basic ' + new Buffer(process.env.ADDRESS_SPACE + ":" + fs.readFileSync(process.env.ADDRESS_SPACE_PASSWORD_FILE)).toString('base64');
     }
-    return result;
-}
 
-function address_path() {
-    var path = "/v1/addresses/";
-    if (process.env.ADDRESS_SPACE) {
-        path += process.env.ADDRESS_SPACE + "/";
+    this.addr_path = "/v1/addresses/";
+    if (this.address_space) {
+        this.addr_path += this.address_space + "/";
     }
-    return path;
 }
 
-var ctrlr = hostport({host:'localhost', port:8080});
-var addr_path = address_path();
-
-function request(path, method, headers, body, handler) {
+AddressCtrl.prototype.request = function (path, method, headers, body, handler) {
+    var self = this;
     return new Promise(function (resolve, reject) {
         var options = {
-            hostname: ctrlr.host,
-            port: ctrlr.port
+            hostname: self.host,
+            port: self.port
         };
         options.path = path;
         options.method = method || 'GET';
+        options.headers = {};
+
         if (headers) {
             options.headers = headers;
         }
 
-        var req = http.request(options, function(res) {
+        if (self.auth_string) {
+            options.headers['Authorization'] = self.auth_string
+        }
+
+        if (self.ca_file) {
+            options.ca = fs.readFileSync(self.ca_file);
```

Comment:
Doing a sync read on every request feels a little wrong in general. I guess you could make the argument that it won't be that frequently called. If we want to re-read it every request perhaps an async read would be nicer. Again not a blocking issue, and something that can also be changed later as needed.